### ISSUE TITLE
Svelte language server issue on imports with neovim's built-in LSP

### ISSUE THREAD

**Issue by charbelnicolas** [neovim/neovim#15805]
### Neovim version (nvim -v)

NVIM v0.6.0-dev+356-gbec7f47ce

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Arch Linux

### Terminal name/version

Kitty

### $TERM environment variable

xterm-kitty

### Installation

AUR

### How to reproduce the issue

https://user-images.githubusercontent.com/14960414/134834083-9842ccb4-cd57-4d04-be53-ef3ffb357b4f.mp4

As requested by Michael, I'm tagging you both on this issue, please check the video @mjlbach and @gpanders. When using the word 'import' and 'from', the error flashes non stop and I get micro freezes (a whole second at the most). It seems the importing of modules is the culprit here.

To reproduce you can open the attached svelte file while having the svelte language server installed and then try to do an import as I did on the video.

[dashboard.svelte.zip](https://github.com/neovim/neovim/files/7232773/dashboard.svelte.zip)

### Expected behavior

No error should show when using imports with the svelte language server and no freezes should happen.

### Actual behavior

When importing modules in svelte files, the word 'import' and the word 'from' cause a lot of glitchiness and small freezes.

**Comment by gpanders** [comment]
Please set the LSP log level to `"debug"`, recreate the issue, and share the log file (`~/.cache/nvim/lsp.log`). It looks like the LSP server is sending invalid diagnostics, but I want to be sure.

**Comment by charbelnicolas** [comment]
Sure, here's the log file:

[lsp.log](https://github.com/neovim/neovim/files/7232887/lsp.log)

**Comment by gpanders** [comment]
The language server is sending a nil value for the line number in its diagnostics. I'll let someone from the LSP team weigh in on this, but I believe this is off-spec behavior and if so the responsibility for fixing it lies with the language server. @neovim/lsp 

```
[DEBUG][2021-09-26 21:12:38] .../vim/lsp/rpc.lua:493	"rpc.receive"	{
    jsonrpc = "2.0",
    method = "textDocument/publishDiagnostics",
    params = {
        diagnostics = {
            {
                code = "parse-error",
                message = 'Unexpected token\n\nIf you expect this syntax to work, here are some suggestions: \nIf you use typescript with `svelte-preprocess`, did you add `lang="typescript"` to your `script` tag? \nDid you setup a `svelte.config.js`? \nSee https://github.com/sveltejs/language-tools/tree/master/docs#using-with-preprocessors for more info.',
                range = {
                    end = {
                        character = 1,
                        line = 3
                    },
                    start = {
                        character = 1,
                        line = 3
                    }
                },
                severity = 1,
                source = "svelte"
            }, {
                code = -1,
                message = "Cannot read properties of undefined (reading 'kind')",
                range = {
                    end = {
                        character = 0,
                        line = vim.NIL
                    },
                    start = {
                        character = 0,
                        line = vim.NIL
                    }
                },
                severity = 1,
                source = "js"
            }
        },
        uri = "file:///home/charbel/Documents/websites/portal-svelte/html/src/routes/dashboard.svelte"
    }
}
```

**Comment by mjlbach** [comment]
It should still provide the line number. I would recommend opening an upstream issue and seeing what the svelte team has to say.

**Comment by mjlbach** [comment]
AFAI can tell should be fixed upstream by https://github.com/sveltejs/language-tools/pull/1183

**Comment by charbelnicolas** [comment]
> AFAI can tell should be fixed upstream by [sveltejs/language-tools#1183](https://github.com/sveltejs/language-tools/pull/1183)

Yep, seems like it, thank you.
